include/geopackage/WKB.hpp

```diff
@@ -0,0 +1,144 @@
+#ifndef NGEN_GEOPACKAGE_WKB_H
+#define NGEN_GEOPACKAGE_WKB_H
+
+#include "EndianCopy.hpp"
+#include "JSONGeometry.hpp"
+#include <boost/geometry/srs/epsg.hpp>
+#include <boost/geometry/srs/transformation.hpp>
+
+namespace bg = boost::geometry;
+
+namespace geopackage {
+
+/**
+ * A recursive WKB reader
```

Comment:
For the benefit of those of us who are new to the project / GIS stuff in general, it would be useful to note that WKB stands for "well-known binary", which is currently specified in an ISO standard.

Also potentially useful to note whether the flavor handled is strictly what's described in the standard, or something like EWKB or some other application's adapted/extended format.

Reply:
One thing to note is that this implementation **partially** follows the OGC specification (see: https://www.ogc.org/standard/sfa/), not the ISO spec (I'm not sure on the differences as I haven't read the ISO version, but most software follows the OGC spec). I'll add a note, but this is a strict WKB (i.e. not EWKB or TWKB) implementation.

I say partially because I only implement the subset of geometry types that we should expect, i.e. the common ones: point, linestring, polygon, and corresponding multi-geometries.

Reply:
Thanks. Those are both details worth recording in the comments

Reply:
Added in 987d3a9.